lib/canvas.js

```diff
@@ -0,0 +1,140 @@
+"use strict";
+/* global require, module, d3 */
+
+var configurable = require('./util/configurable');
+
+module.exports = function (d3) {
+
+	return function (config) {
+
+    config = config || {
+      xScale: null,
+      eventColor: null
+    };
+    for (var key in defaultConfig) {
+      config[key] = config[key] || defaultConfig[key];
+    }
+
+	var canvasHandler = function canvasHandler() {
+		var canvas = document.getElementsByTagName('canvas')[0];
+	canvas.width = 800; canvas.height = 600;
+	
+	window.onload = function(){		
+		var ctx = canvas.getContext('2d');
+		trackTransforms(ctx);
+		function redraw(){
+			// Clear the entire canvas
+			var p1 = ctx.transformedPoint(0,0);
+			var p2 = ctx.transformedPoint(canvas.width,canvas.height);
+			ctx.clearRect(p1.x,p1.y,p2.x-p1.x,p2.y-p1.y);
+
+			ctx.font = "30px Arial";
+          ctx.textAlign = "center";
+          ctx.fillText("Toto",750/2,380/2);
+		}
+		redraw();
+		
+		var lastX=canvas.width/2, lastY=canvas.height/2;
+		var dragStart,dragged;
+		canvas.addEventListener('mousedown',function(evt){
+			document.body.style.mozUserSelect = document.body.style.webkitUserSelect = document.body.style.userSelect = 'none';
+			lastX = evt.offsetX || (evt.pageX - canvas.offsetLeft);
+			lastY = evt.offsetY || (evt.pageY - canvas.offsetTop);
+			dragStart = ctx.transformedPoint(lastX,lastY);
+			dragged = false;
+		},false);
+		canvas.addEventListener('mousemove',function(evt){
```

Comment:
Just like you did with the `zoom` function, it'd be a nice idea to extract the callback.